test/verify/check-system-info

```diff
@@ -559,33 +507,49 @@ revision\t: 2.3 (pvr 004e 1203)
         b.enter_page('/system/hwinfo')
         b.wait_in_text('#hwinfo-system-info-list .hwinfo-system-info-list-item:nth-of-type(2) .pf-c-description-list__group:nth-of-type(1) dd', "2x POWER9 (architected), altivec supported")
 
-        # Memory details should be shown for generic QEMU memory
-        # demidecode not available on certain images
+        # Memory details should be shown from our mocked DMI information from systemd's test files.
         if not m.ostree_image:
             b.wait_in_text('#hwinfo #memory-listing' + heading_selector, "Memory")
             b.wait_in_text('#hwinfo #memory-listing table', "DIMM")
             b.wait_in_text('#hwinfo #memory-listing table', "RAM")
 
-            # Test more specific memory data with a fake dmidecode
-            m.write('/tmp/dmidecode', dmidecode)
-            m.execute('chmod +x /tmp/dmidecode; mount -o bind /tmp/dmidecode $(which dmidecode)')
-            self.addCleanup(m.execute, 'until umount $(which dmidecode); do sleep 1; done')
+            m.upload(["verify/files/dmi/smbios_entry_point", "verify/files/dmi/Lenovo-ThinkPad-X280.bin"], "/tmp/")
+            m.execute("mount -o bind /tmp/smbios_entry_point /sys/firmware/dmi/tables/smbios_entry_point")
+            self.addCleanup(m.execute, "umount /sys/firmware/dmi/tables/smbios_entry_point")
+            m.execute("mount -o bind /tmp/Lenovo-ThinkPad-X280.bin /sys/firmware/dmi/tables/DMI")
+            self.addCleanup(m.execute, "umount /sys/firmware/dmi/tables/DMI")
+            # umount dmi overlay, otherwise we can't retrigger udev rules
+            m.execute("umount /sys/class/dmi/id")
+            m.execute("udevadm trigger --verbose /sys/devices/virtual/dmi/id")
+
             b.reload()
             b.enter_page('/system/hwinfo')
-            # first DIMM has unknown technology and rank
+            systemd_version = int(m.execute("systemctl --version | awk '{print $2; exit}'").strip())
+
+            # Test more specific memory data with a fake dmidecode
             b.wait_in_text('#memory-listing tr:nth-of-type(1) td[data-label=ID]', "BANK 0: ChannelA-DIMM0")
-            b.wait_in_text('#memory-listing tr:nth-of-type(1) td[data-label=Type]', "DDR3")
-            b.wait_in_text('#memory-listing tr:nth-of-type(1) td[data-label=Size]', "8 GiB")
+            b.wait_in_text('#memory-listing tr:nth-of-type(1) td[data-label=Type]', "DDR4")
+            # dmidecode shows GB, ubuntu backports changes to systemd 245
+            if systemd_version < 248 and m.image != "ubuntu-2004":
```

Comment:
Such a mix is ugly IMHO -- it's nicer to just have a list of "old" OSes that shrinks to zero over time, and an OS list is much easier to grep for. E.g. once we drop testing rhel-8-6 or ubuntu-2004, we grep for these OS image names.